components/plurals/src/rules/runtime/resolver.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::operands::PluralOperands;
+use crate::rules::runtime::ast;
+
+pub fn test_rule(rule: &ast::Rule, operands: &PluralOperands) -> bool {
+    let mut left = true;
+
+    for relation in rule.0.iter() {
+        let relation = relation.as_relation();
+        if left {
+            if relation.and_or == ast::AndOr::Or {
+                return true;
+            } else {
+                left = test_relation(&relation, operands);
+            }
+        } else if relation.and_or == ast::AndOr::Or {
+            left = test_relation(&relation, operands);
+        }
+    }
+    left
+}
+
+#[inline]
+fn test_relation(relation: &ast::Relation, operands: &PluralOperands) -> bool {
+    let result = if let Some(value) = get_value(relation, operands) {
+        relation.range_list.iter().any(|range| match range {
+            ast::RangeOrValue::Value(v) => u64::from(v) == value,
+            ast::RangeOrValue::Range(min, max) => {
+                value >= u64::from(min) && value <= u64::from(max)
+            }
+        })
+    } else {
+        false
+    };
+    match relation.polarity {
+        ast::Polarity::Negative => !result,
+        ast::Polarity::Positive => result,
+    }
+}
+
+#[inline]
+fn get_value(relation: &ast::Relation, operands: &PluralOperands) -> Option<u64> {
```

Comment:
Nit (optional): I think it might be more clear if you inlined `get_value` into `test_relation` in the source code.

Reply:
I tried it and didn't like it. It made the code less readable to me and had to use more Option methods to work around the lack of `return None`. I prefer separated with inline.